src/test/python/apache/aurora/config/test_loader.py

```diff
@@ -114,6 +160,59 @@ def test_load():
       assert hello_world.name().get() == 'hello_world'
 
 
+def test_load_with_includes():
+  with temporary_dir() as tmp_dir:
+    f1_name = 'f1.aurora'
+    f2_name = 'f2.aurora'
+    with open(os.path.join(tmp_dir, f1_name), 'w+') as f1:
+      f1.write(MESOS_CONFIG)
+      f1.flush()
+      f1.seek(0)
+      with open(os.path.join(tmp_dir, f2_name), 'w+') as f2:
+        f2.write(MESOS_CONFIG_WITH_INCLUDE_TEMPLATE % f1_name)
+        f2.flush()
+        f2.seek(0)
+
+        env = AuroraConfigLoader.load(f2.name, is_memoized=True)
+        assert 'jobs' in env and len(env['jobs']) == 2
+        hello_world = env['jobs'][0]
+        assert hello_world.name().get() == 'hello_world'
+        other_job = env['jobs'][1]
+        assert other_job.name().get() == 'otherjob'
+
+
+def test_memoized_load():
+  with temporary_dir() as d:
+    with open(os.path.join(d, 'config.aurora'), 'w+') as fp:
+      fp.write(MESOS_CONFIG)
+      fp.flush()
+      fp.seek(0)
+
+      for config in (fp.name, fp):
+        env = AuroraConfigLoader.load(config, is_memoized=True)
+        assert 'jobs' in env and len(env['jobs']) == 1
+        hello_world = env['jobs'][0]
+        assert hello_world.name().get() == 'hello_world'
+
+    with open(os.path.join(d, 'config.aurora'), 'w+') as fp:
+      fp.write(MESOS_CONFIG_MULTI)
+      fp.flush()
+      fp.seek(0)
+
+      for config in (fp.name, fp):
+        # Verfiy Cached Content is from initial write/read, (1 job)
+        env = AuroraConfigLoader.load(config, is_memoized=True)
+        assert 'jobs' in env and len(env['jobs']) == 1
```

Comment:
It is not clear to me how we are testing the memoization. Should we inject `CACHED_ENV` and assert it does not get updated when using memoization instead?